rmf_api_msgs/samples/task_state/multi_dropoff_delivery.json

```diff
@@ -0,0 +1,214 @@
+{
+  "booking": {
+    "id": "delivery_2021:11:08:23:50",
+    "unix_millis_earliest_start_time": 1636388400000,
+    "priority": "none",
+    "automatic": false
+  },
+  "header": {
+    "category": "Multi-Delivery",
+    "detail": [
+      {
+        "category": "Pick Up",
+        "params": {
+          "location": "Kitchen",
+          "items": [
+            {
+              "type": "soda",
+              "quantity": 1
+            },
+            {
+              "type": "water",
+              "quantity": 1
+            }
+          ]
+        }
+      },
+      {
+        "category": "Drop Off",
+        "params": {
+          "location": "room_203",
+          "items": [
+            {
+              "type": "soda",
+              "quantity": 1
+            }
+          ]
+        }
+      },
+      {
+        "category": "Drop Off",
+        "params": {
+          "location": "room_521",
+          "items": [
+            {
+              "type": "water",
+              "quantity": 1
+            }
+          ]
+        }
+      }
+    ],
+    "estimate_millis": 2000000
+  },
+  "completed": [
+    {
+      "phase": {
+        "id": 1,
+        "header": {
```

Comment:
I feel that some/all of the fields in `header` makes more sense at the top level. For example `phase.category` feels more natural than `phase.header.category`, the latter implies it is a category of the header and not the category of the phase.

Reply:
The `(category, detail, time estimate)` triplet is shared by both the task schema and the phase schema, so I decided to group them into the `header` schema to easily share them between task and phase. We can certainly flatten the structures if you feel strongly about it, though.

Reply:
Personally I prefer them at the top level, but these are just preference and neither are more "correct". So let's see what other people think.

Reply:
You're right that it's an unnecessary layer. The small amount of duplication in the data structures isn't important; I'll go ahead and move the fields up a level and get rid of `header`.

Reply:
I think the case of duplicating fields is that because json schemas are usually generated from source code. Interestingly, the latest draft suggests that values adjacent to "$ref" are allowed, but it's up to the application to decide how to interpret it. In practice, this is probably useless as it makes it so that the schema itself is not sufficient to describe the structure of the data (https://json-schema.org/draft/2020-12/json-schema-core.html#rfc.section.7.7.1.1).

Reply:
I've flattened out the header structure here: 45ab009

I'm going to avoid using an adjacent "$ref" for now because of the ambiguity that you mentioned.

Reply:
https://stackoverflow.com/questions/61678570/override-a-inherited-json-property-with-a-specific-value It seems we can inherit and extend schemas, it also suggests using adjacent keywords next to `$ref`, perhaps there is some default rules wrt to how to merge the fields, or there is a industry standard which most validators use.

Reply:
I think the three fields are modest enough that the complexity and confusion of inheritance isn't really worth it.

Reply:
The header is removed now.